preview/action.yml

```diff
@@ -0,0 +1,71 @@
+name: 'Branch Preview'
+description: 'Preview Fastly Service'
+
+inputs:
+  fastly-api-token:
+    description: 'The Fastly API token to use for interacting with Fastly API'
+    required: true
+  github-token:
+    description: 'The GitHub token to use for downloading the Fastly CLI'
+    required: true
+
+outputs:
+  domain:
+    description: "Preview domain for Fastly Compute@Edge Service"
+    value: ${{ steps.domain.outputs.DOMAIN }}
+  service_name:
+    description: "Preview service name for Fastly Compute@Edge Service"
+    value: ${{ steps.service-name.outputs.SERVICE_NAME }}
+
+runs:
+    using: "composite"
+    steps:
+      - uses: actions/checkout@v3
+      # Download Fastly CLI
+      - name: Set up Fastly CLI
+        uses: fastly/compute-actions/setup@v4
+        with:
+          token: ${{ inputs.github-token }}
+          cli_version: '10.2.0'
+      - run: yarn
+        shell: bash
+
+      # Create a new Fastly Service name with the PR number appended
+      - name: Set service-name
+        id: service-name
+        run: echo "SERVICE_NAME=$(yq '.name' fastly.toml)-${{ github.event.number }}" >> "$GITHUB_OUTPUT"
+        shell: bash
+
+      # Delete the Fastly Service
+      - if: github.event.action == 'closed'
+        run: fastly service delete --quiet --service-name ${{ steps.service-name.outputs.SERVICE_NAME }} --force --token ${{ inputs.fastly-api-token }} || true
+        shell: bash
+
+      # Add the service name to fastly.toml so that the Fstly CLI is aware of the new name
+      - if: github.event.action != 'closed'
+        run: sed -i 's/^name = ".*"/name = "${{ steps.service-name.outputs.SERVICE_NAME }}"/g' fastly.toml
+        shell: bash
+      # Deploy to Fastly and let Fastly choose a subdomain of edgecompute.app to attach to the service
+      - if: github.event.action != 'closed'
+        run: |
+          if fastly service describe --quiet --service-name="${{ steps.service-name.outputs.SERVICE_NAME }}" --token ${{ inputs.fastly-api-token }} ; then
+            FASTLY_SERVICE_ID="$(fastly service describe --service-name="${{ steps.service-name.outputs.SERVICE_NAME }}" --token ${{ inputs.fastly-api-token }} --quiet --json | jq -r '.ID')" fastly compute publish --verbose --token ${{ inputs.fastly-api-token }}
+          else
+            fastly compute publish --verbose -i --token ${{ inputs.fastly-api-token }}
+          fi
+        shell: bash
```

Comment:
If the Fastly CLI could be changed to accept `--service-name` as a flag on `fastly compute publish`, then all these lines could be replaced with:
```
fastly compute publish --verbose -i --token ${{ inputs.fastly-api-token }} --service-name ${{ steps.service-name.outputs.SERVICE_NAME }}

@Integralist what do you think to this idea?

Reply:
copying in @Integralist for visibility

Reply:
@JakeChampion @kailan 

Looks like `--service-name` is already exposed for `fastly compute publish`:

https://developer.fastly.com/reference/cli/compute/publish/

Reply:
Yup...

https://github.com/fastly/cli/blob/main/pkg/commands/compute/deploy.go#L230-L233

The `cmd.ServiceID` function first tries to get a Service ID from a `--service-id` flag, then from the `FASTLY_SERVICE_ID` environment variable, then from the fastly.toml manifest file. If all of those fail, it will check if `--service-name` was set and if so it will list all services and try to find one that matches the service name value.

> **IMPORTANT:** I fixed a bug in that [service name matching code](https://github.com/fastly/cli/blob/main/pkg/cmd/flags.go#L157C6-L174) yesterday, but I've not yet published it into a new release (I'm doing that today). The bug was that we weren't paginating the list of services so we only checked the first page of services.

Reply:
@Integralist That's correct, however it is the last part which is the issue:
>it will check if --service-name was set and if so it will list all services and try to find one that matches the service name value.

As this is a new service being created, that logic will find no such service and then I believe it fails to create a new service.

I just checked with the commands below and it errors:
```
❯ head fastly.toml
# This file describes a Fastly Compute@Edge package. To learn more visit:
# https://developer.fastly.com/reference/fastly-toml/

authors = ["me@jakechampion.name"]
description = "hn"
language = "javascript"
manifest_version = 2
name = "hn"
service_id = ""

❯ fastly compute publish --profile personal --verbose --service-name meow && fastly log-tail --profile personal
Fastly API token provided via config file (profile: personal)
Fastly API endpoint: https://api.fastly.com

✓ Verifying fastly.toml
✓ Identifying package name
✓ Identifying toolchain

INFO: Creating ./bin directory (for Wasm binary)

Build script to execute:
	sh -c npm run build


INFO: Command output:
--------------------------------------------------------------------------------

> hackercat@1.0.0 build
> js-compute-runtime src/index.js bin/main.wasm

--------------------------------------------------------------------------------

✓ Running [scripts.build]
✓ Creating package archive

SUCCESS: Built package (pkg/hn.tar.gz)


ERROR: error listing service versions: missing required field 'ServiceID'.

If you believe this error is the result of a bug, please file an issue: https://github.com/fastly/cli/issues/new?labels=bug&template=bug_report.md
```

Reply:
Oh, so you're talking about a scenario where you're creating a new service, and so you want the `--service-name` flag to be used as the name of the service to be created, is that right?

OK, then yes in that case the CLI doesn't handle that scenario as the flag is intended to be used as a 'lookup' (as an alternative to remembering a service id) and not to be used as a 'value'.

I'll create a ticket for this

Reply:
>Oh, so you're talking about a scenario where you're creating a new service, and so you want the --service-name flag to be used as the name of the service to be created, is that right?

That's right, yes 👍

Reply:
https://github.com/fastly/cli/pull/979

Reply:
```suggestion
      # Deploy to Fastly and let Fastly choose a subdomain of edgecompute.app to attach to the service
      - if: github.event.action != 'closed'
        run: |
          fastly compute publish --verbose -i --token ${{ inputs.fastly-api-token }} --service-name ${{ steps.service-name.outputs.SERVICE_NAME }}
        shell: bash
```

Reply:
FYI that PR was merged but I've a bunch of other CLI stuff I need to work on first before cutting a new release. But I'll let y'all know when the next release is out.

Reply:
@Integralist do you know when the ETA for the latest CLI is? Would love to get use the preview action asap!

Reply:
@gnohj I'm aiming for getting a new release out before the end of next week

Reply:
Excited to see this come down the pipe !!

Reply:
@kailan https://github.com/fastly/cli/releases/tag/v10.2.4